server/writer/gzip.go

```diff
@@ -86,6 +88,11 @@ func (g *Gzip) Hijack() (net.Conn, *bufio.ReadWriter, error) {
 	return hijack.Hijack()
 }
 
+func (g *Gzip) SkipCompression() {
```

Comment:
Would it be possible, that the gzip writer decides to skip the compression or not? Now we have at least 3 different places which calling this method or setting related context values.
Additionally is the skip bool flag obsolete? Since `enabled` is set to `false` within the same step.